File: EasyPlan.Web/Components/Mapper/ItemMapper.cs
using System.Linq;
using EasyPlan.DomainModel.Entities;

namespace EasyPlan.Web.Components.Mapper
{
    public static class ItemMapper
    {
        public static object Map(Item entity)
        {
            return new
            {
                id = entity.Id,
                title = entity.Title,
                marks = entity.Marks.Select(e => MarkMapper.Map(e))
            };
        }
    }
}

File: EasyPlan.Web/App/mappers/itemMapper.js
define(['models/item', 'mappers/markMapper'],
    function (Item, markMapper) {
    return {
        map: map,
    }

    function map(src) {
        return new Item({
            id: src.id,
            title: src.title,
            marks: _.map(src.marks, markMapper.map)
        })
    }   
});